Formula/python@3.9.rb

```diff
@@ -130,6 +163,19 @@ def install
     # Avoid linking to libgcc https://mail.python.org/pipermail/python-dev/2012-February/116205.html
     args << "MACOSX_DEPLOYMENT_TARGET=#{MacOS.version}"
 
+    # Python's setup.py parses CPPFLAGS and LDFLAGS to learn search
+    # paths for the dependencies of the compiled extension modules.
+    # See Linuxbrew/linuxbrew#420, Linuxbrew/linuxbrew#460, and Linuxbrew/linuxbrew#875
+    on_linux do
+      if build.bottle?
+        # Configure Python to use cc and c++ to build extension modules.
```

Comment:
/cc @Homebrew/linux

Reply:
This was added by this PR for python 2.7 in 2016: https://github.com/Homebrew/linuxbrew-core/pull/483

I created a test build here https://github.com/Homebrew/linuxbrew-core/pull/21744 to see what happens without this hack.

Reply:
I'm digging back in my memory here, but I believe those shim compiler scripts then end being the compiler that's used by Python to build native extension modules, even outside of Homebrew, and those shim compiler scripts don't work outside of Homebrew. I may be wrong, since it seems to work on macOS, which I would expect to have the same issue.

Reply:
I made a test without resetting ENV["CC"], ENV["CXX"]. From a pure building point of view, the build passes.

I think the comment in the linux formula is a little bit misleading:
```
# Python's setup.py parses CPPFLAGS and LDFLAGS to learn search
# paths for the dependencies of the compiled extension modules.
# See Homebrew/linuxbrew#420, Homebrew/linuxbrew#460, and Homebrew/linuxbrew#875
```

This is related to the ldflags and ccpflags only, and not to the ENV["CC"], ENV["CXX"] part. It was just added in the same block for convenience I guess.

I would like to get rid of the ENV["CC"], ENV["CXX"] lines, and see if something breaks. Then we will have a real clue what is really broken.

Reply:
I believe the thing you need to test is compiling a native Python extension module with the resulting Python installation. Something like `igraph` perhaps.

Reply:
I tested a `igraph` build in a clean docker container, with python built without the ENV hacks, and it worked. Not sure what I can test.
I also ran `brew audit` on python@3.9, and I think it would have complained about shim paths, but it did not.

Reply:
Good enough for me!

Reply:
@SeekingMeaning You can just remove the whole `on_linux do` block then.